src/Drivers/CvrApi.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Bilfeldt\VatService\Drivers;
+
+use Bilfeldt\VatService\Exceptions\DriverUnavailable;
+use Bilfeldt\VatService\Exceptions\InvalidVatException;
+use Bilfeldt\VatService\Exceptions\UnsupportedCountryException;
+use Bilfeldt\VatService\Exceptions\VatNotFoundException;
+use Bilfeldt\VatService\VatInformation;
+use Bilfeldt\VatService\VatServiceInterface;
+use Illuminate\Http\Client\Response;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Http;
+
+class CvrApi implements VatServiceInterface
+{
+    private static array $formats = [
+        'DK' => [
+            '########'
+        ],
+        'NO' => [
+            // empty means any format is accepted
+        ],
+    ];
+
+    public static function supports(string $countryCode): bool
+    {
+        return array_key_exists(strtoupper($countryCode), self::$formats);
+    }
+
+    public function __construct(
+        private ?string $accessToken, // This api actually works without an access token, but then rate limiting is applied.
+    ) {}
+
+    /** @inheritDoc */
+    public function getFormats(string $countryCode): Collection
+    {
+        if (! self::supports($countryCode)) {
+            throw UnsupportedCountryException::unsupported($countryCode);
+        }
+
+        return Collection::make(self::$formats[strtoupper($countryCode)]);
+    }
+
+    /** @inheritDoc */
+    public function isValidFormat(string $countryCode, string $vatNumber): bool
+    {
+        if (! self::supports($countryCode)) {
+            throw UnsupportedCountryException::unsupported($countryCode);
+        }
+
+        return $this
+            ->getFormats($countryCode)
+            ->whenEmpty(
+                fn (Collection $collection) => true,
+                function (Collection $collection) use ($vatNumber): bool {
+                    return $collection->filter(function (string $format) use ($vatNumber) {
+                        return true; // TODO: Fix this
+                    })->isNotEmpty();
+                }
+            );
+    }
+
+    /** @inheritDoc */
+    public function isValid(string $countryCode, string $vatNumber): bool
+    {
+        if (! $this->isValidFormat($countryCode, $vatNumber)) {
+            return false;
+        }
```

Comment:
@bilfeldt should we not throw `InvalidVatException` here as it is invalid format instead of returning false? same with other two exception down the line.

Should we not throw exception here and let the package consumer handle the exception how they want to response to those exception in their app as there are 3 different scenarios are at play here?

Reply:
No the point here is that if the vat number has an invalid format, then we **know** that it is wrong, and we should return `false`.

This function should only throw an exception if it is not capable of determining **if** the vat number is valid or not, which can only happen because of two things:

1. Problems with the driver - those exceptions are driver specific and should bobble up.
2. The vat number was not found **and** the driver is not complete, so we do not know if the number is in fact invalid or if it is just missing from the drivers dataset.

Makes sense?